app/controllers/application_controller.rb

```diff
@@ -18,7 +18,10 @@ def permission_check
 
   # Check if logged in, and if not, then redirect
   def check_if_logged_on
-    if request.path != "/login"
+    #get the current path and verify that the users are logged in, or trying to login/signup
+    controller = params[:controller]
+    action = params[:action]
+    if request.path != "/login" && request.path != "/signup" && !(controller == "users" && action == "create")
```

Comment:
Is it possible to have a whitelist for links that can be accessed without being logged in? Maybe an array or something? In the future, we might add other links like Landing page, about page, contact page, etc. It's not that important but maybe something to think about